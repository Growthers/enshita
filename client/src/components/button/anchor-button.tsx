import type { FC } from "react";
import { InlineIcon } from "@iconify/react";
import styles from "~/styles/components/button/anchor-button.module.scss";
import type {
  ButtonIconProperties,
  AnchorButtonProperties,
} from "./type/model";

const ButtonIcon: FC<ButtonIconProperties> = ({ icon, css, ...rest }) => (
  <div className={`${styles["enshita-button-icon-box"]} ${css}`}>
    <InlineIcon
      icon={icon}
      className={styles["enshita-button-icon"]}
      {...rest}
    />
  </div>
);

const AnchorButton: FC<AnchorButtonProperties> = ({
  children,
  icon,
  variant,
  iconStyles,
  boxStyles,
  textStyles,
  isSamePage,
  href,
  ...rest
}) => (
  <a
    className={`${styles["enshita-button-box"]} ${boxStyles}`}
    href={href}
    target={isSamePage ? "" : "_blank"}
    rel="noopener noreferrer"
    role="button"
    {...rest}
  >
    {variant === "icon" && <ButtonIcon icon={icon} css={iconStyles} />}
    {children && (
      <p className={`${styles["enshita-button-text"]} ${textStyles}`}>
        {children}
      </p>
    )}
  </a>
);

export { AnchorButton };
