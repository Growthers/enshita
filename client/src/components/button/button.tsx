import type { FC } from "react";
import { InlineIcon } from "@iconify/react";
import styles from "~/styles/button.module.scss";
import type { ButtonIconProperties, ButtonProperties } from "./type/model";

const ButtonIcon: FC<ButtonIconProperties> = ({ icon, css, ...rest }) => (
  <div className={`${styles["enshita-button-icon-box"]} ${css}`}>
    <InlineIcon
      icon={icon}
      className={styles["enshita-button-icon"]}
      {...rest}
    />
  </div>
);

/* eslint-disable react/button-has-type */
const Button: FC<ButtonProperties> = ({
  children,
  icon,
  variant,
  disabled,
  iconStyles,
  boxStyles,
  textStyles,
  type,
  ...rest
}) => (
  <button
    className={`${styles["enshita-button-box"]} ${boxStyles}`}
    disabled={disabled}
    tabIndex={0}
    type={type}
    {...rest}
  >
    {variant === "icon" && <ButtonIcon icon={icon} css={iconStyles} />}
    {children && (
      <p className={`${styles["enshita-button-text"]} ${textStyles}`}>
        {children}
      </p>
    )}
  </button>
);
/* eslint-enable react/button-has-type */

export { Button };
