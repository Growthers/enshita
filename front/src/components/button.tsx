import { FC, ReactNode } from "react";
import styles from "~/styles/button.module.scss";

type Props = {
  children: string | ReactNode;
  onClick?: () => void;
  txColor?: string;
  bgColor?: string;
  style?: string;
  buttonType?: JSX.IntrinsicElements["button"]["type"];
};

const Button: FC<Props> = ({
  children,
  onClick = () => {},
  txColor = "enshita-button-color-brack",
  bgColor = "enshita-button-bg-secondary",
  style = "enshita-button-normal",
  buttonType = "button",
}) => (
  <button
    // eslint-disable-next-line react/button-has-type
    type={buttonType}
    onClick={onClick}
    className={`
      ${styles[txColor]}
      ${styles[bgColor]}
      ${styles[style]}
      `}
  >
    {children}
  </button>
);

Button.defaultProps = {
  onClick: () => {},
  txColor: "enshita-button-color-brack",
  bgColor: "enshita-button-bg-secondary",
  style: "enshita-button-normal",
  buttonType: "button",
};

export default Button;
