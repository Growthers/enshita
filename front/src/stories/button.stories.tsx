import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Button from "~/components/button";

type T = typeof Button;
type Story = ComponentStoryObj<T>;

export default {
  title: "Button",
  args: { children: <div>hogehoge</div> },
  component: Button,
} as ComponentMeta<T>;

export const Default: Story = {};
