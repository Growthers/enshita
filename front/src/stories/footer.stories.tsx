import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Footer from "~/components/footer";

type T = typeof Footer;
type Story = ComponentStoryObj<T>;

export default {
  title: "Footer",

  component: Footer,
} as ComponentMeta<T>;

export const Default: Story = {};
