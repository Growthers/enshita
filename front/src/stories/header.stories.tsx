import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStory } from "@storybook/react";
import Header from "~/components/header";

type T = typeof Header;
type Story = ComponentStory<T>;

export default {
  title: "Header",

  component: Header,
} as ComponentMeta<T>;

export const Default: Story = {};
