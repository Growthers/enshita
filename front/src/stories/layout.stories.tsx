import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStory } from "@storybook/react";
import Layout from "~/components/layout";

type T = typeof Layout;
type Story = ComponentStory<T>;

export default {
  title: "Layout",
  args: { children: <div>hogehoge</div> },
  component: Layout,
} as ComponentMeta<T>;

export const Default: Story = {};
