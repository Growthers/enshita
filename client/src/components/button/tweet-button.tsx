import React from "react";
import { AnchorButton } from "~/components/button/anchor-button";
import styles from "~/styles/components/button/tweet-button.module.scss";

type Props = {
  text: string;
  url: string | undefined;
  hashtags: string[] | undefined;
};

const TweetButton: React.FC<Props> = ({ text, url = "", hashtags = [] }) => {
  // URL, hashtagsがないときはパラメータの値がなにもないが、問題なく指定した項目のみでTweet可能
  const tweetLink = `https://twitter.com/intent/tweet?text=${text.replace(
    / /g,
    "%20",
  )}&url=${url}&hashtags=${
    hashtags ? hashtags.map(v => v.replace("#", "")).join(",") : ""
  }`;

  return (
    <AnchorButton
      variant="icon"
      href={tweetLink}
      icon="akar-icons:twitter-fill"
      iconStyles={styles["enshita-event-tweet-icon"]}
      boxStyles={styles["enshita-event-tweet-box"]}
      textStyles={styles["enshita-event-tweet-text"]}
    >
      Tweet
    </AnchorButton>
  );
};

export default TweetButton;
