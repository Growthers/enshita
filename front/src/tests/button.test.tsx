import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "~/stories/button.stories";

const { Default } = composeStories(stories);

describe("(components) footer", () => {
  test("test Snap Shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
