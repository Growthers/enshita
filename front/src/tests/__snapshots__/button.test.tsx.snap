// Vitest Snapshot v1

exports[`(components) footer > test Snap Shot 1`] = `
<div>
  <button
    class="
      _enshita-button-color-brack_1mkh6_5
      _enshita-button-bg-secondary_1mkh6_1
      _enshita-button-normal_1mkh6_9
      "
    type="button"
  >
    <div>
      hogehoge
    </div>
  </button>
</div>
`;
